jinahub/rankers/LightGBMRanker/lightgbm_ranker.py

```diff
@@ -0,0 +1,209 @@
+__copyright__ = "Copyright (c) 2020-2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import os
+from typing import Dict, Optional, List
+
+import lightgbm
+import numpy as np
+from jina.logging.logger import JinaLogger
+from jina import Executor, DocumentArray, requests
+
+
+class LightGBMRanker(Executor):
+    """
+    Computes a relevance score for each match using a pretrained Ltr model trained with LightGBM (https://lightgbm.readthedocs.io/en/latest/index.html)
+    :param query_features: name of the features to extract from query Documents and used to compute relevance scores by the model loaded
+    from model_path
+    :param match_features: name of the features to extract from match Documents and used to compute relevance scores by the model loaded
+    from model_path
+    :param relevance_label: If call :meth:`train` endpoint, the label will be used as groundtruth for model training. If on :meth:`rank` endpoint, the
+    label will be used to assign a score to :attr:`Document.scores` field.
+    :param model_path: path to the pretrained model previously trained using LightGBM.
+    :param params: Parameters used to train the LightGBM learning-to-rank model.
+    :param categorical_query_features: name of features contained in `query_features` corresponding to categorical features.
+    :param categorical_match_features: name of features contained in `match_features` corresponding to categorical features.
+    :param query_features_before: True if `query_features` must be placed before the `match` ones in the `dataset` used for prediction.
```

Comment:
Is this really needed? I think we can just establish a convention, note it, and that's it. If we train the model in the executor it doesn't matter, if it is trained outside the users can easily set the proper order of features.

Reply:
Let's keep it like this and maybe change it later.